pytensor/tensor/rewriting/subtensor.py

```diff
@@ -1864,21 +1865,18 @@ def local_uint_constant_indices(fgraph, node):
     if not has_new_index:
         return False
 
-    new_out = x[tuple(new_indices)]
-
-    if y is not None:
-        new_out = inc_subtensor(
-            new_out,
-            y,
-            inplace=node.op.inplace,
-            set_instead_of_inc=node.op.set_instead_of_inc,
-            ignore_duplicates=getattr(node.op, "ignore_duplicates", False),
-        )
-
-    new_outs = new_out.owner.outputs
-    copy_stack_trace(node.outputs, new_outs)
-
-    return new_outs
+    if isinstance(op, Subtensor | IncSubtensor):
+        # Basinc index Ops contain information about the dtype of the indices, so wee have to recreate them
+        props = op._props_dict()
+        props["idx_list"] = new_indices
+        op = type(op)(**props)
+        # Basic index Ops don't expect slices, but the respective start/step/stop
+        new_indices = get_slice_elements(new_indices)
+
+    new_args = (x, *new_indices) if y is None else (x, y, *new_indices)
+    new_out = op(*new_args)
```

Comment:
This avoids a regression where AdvancdeSubtensor1 introduced previously would be replaced by an AdvancdeSubtensor again (and require a useless specify_shape, due to loss of static type info).

Reply:
It made an infinite loop?

Reply:
No, this rewrite was only applied once (once the dtypes become specialized this rewrite no longer does anything), but the one that introduces AdvancedSubtensor1 was needed twice, because it was undone by this one.

Calling `inc_subtensor` or `x[idx]` never introduces AdvancedSubtensor1 directly.